onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -85,6 +84,38 @@ Status ValidateInputs(const Tensor* depth,
   return Status::OK();
 }
 
+Status PrepareOutputShape(const Tensor* indices, const int64_t depth_val, const int64_t axis,
+                          int64_t& prefix_dim_size, int64_t& suffix_dim_size,
+                          std::vector<int64_t>& output_shape) {
+  const auto& indices_shape = indices->Shape();
+  const auto& indices_dims = indices_shape.GetDims();
+  const auto indices_num_dims = indices_shape.NumDimensions();
+  output_shape = indices_dims;
+
+  // output rank is always 1 more than the input rank as a new dimension is added to the input shape
+  const auto output_rank = static_cast<int64_t>(indices_num_dims + 1);
+  if (axis >= output_rank || axis < -output_rank) {
+    std::ostringstream oss;
+    oss << "'axis' attribute must have a value in the range [" << -output_rank
+        << "," << indices_num_dims << "]";
+    return Status(ONNXRUNTIME, INVALID_ARGUMENT, oss.str());
+  }
+
+  auto true_axis = axis;
+  if (true_axis < 0)
+    true_axis += output_rank;
+
+  output_shape.insert(output_shape.begin() + true_axis, depth_val);
+
+  prefix_dim_size = 1;
+  for (int64_t i = 0; i < true_axis; ++i) {
+    prefix_dim_size *= indices_dims[i];
```

Comment:
>indices_dims [](start = 23, length = 12)

could this be scalar? Then indices_dims is 0-size

Reply:
No. Spec Shape Inference enforces this 
" Indices tensor must have rank >= 1"